packages/common/browser/track-leaders.vue

```diff
@@ -0,0 +1,39 @@
+<template>
+  <div class="common-track-leaders" />
+</template>
+
+<script>
+import { get } from 'object-path';
+
+const getAsObject = (obj, path) => {
+  const v = get(obj, path, {});
+  return v && typeof v === 'object' ? v : {};
+};
+
+export default {
+  inject: ['EventBus'],
+  props: {
+    name: {
+      type: String,
+      default: 'dataLayer',
+    },
+    event: {
+      type: String,
+      default: 'leaders-action',
+    },
+  },
+  mounted() {
+    this.EventBus.$on('leaders-action', (event, payload) => {
+      const { type, label, category } = getAsObject(event);
+      const data = {
+        type,
+        label,
+        category,
+        ...getAsObject(payload),
+      };
+      const dataLayer = window[this.name];
+      if (dataLayer) dataLayer.push({ ...data, event: this.event });
```

Comment:
Also, wrap _all_ of the leaders data into a `leaders`  property, instead of spreading it to the root. That way you won't have potential conflicts or overridden data.